### ISSUE TITLE
Replace webpack with bun?

### ISSUE THREAD

**Issue by phtmgt** [symfony/webpack-encore#1222]
Just a conversation starter. Bun seems to be all the rage, so I think it’s the natural next step. It’s Nx faster and simpler than everything else and I am looking forward to the day when webpack won’t be taking ages to (re)compile.

**Comment by stof** [comment]
Given that Encore is primarily a builder for the webpack config, we cannot decouple it from webpack (that would be a new project instead).
However, the good news is that the Symfony framework does not couple itself to webpack or to encore. If you have a builder outputting files, you can use them with the asset component. If you want to perform code splitting of entrypoints, you could even output a JSON file in the same format than Encore and the WebpackEncoreBundle could still be used.
The trickiest part would probably be to implement the integration with Symfony UX to load the stimulus controllers registered by Flex.

In Symfony 6.3, we even introduced a different way to manage assets that does not rely on webpack, for projects targeting modern browsers.

Side note: when I read the Bun website, I have the impression that it is a replacement for node.js for backend JS code, not a bundler for frontend JS code.

**Comment by weaverryan** [comment]
Probably encore should be retired for people that still want to use a build system in favor of vite or something else. Not sure about bun - as Stof said, it's a replacement for node.js, not a build system precisely. We'll see how things shake out

**Comment by phtmgt** [comment]
As far as I know, Bun is a [bun]dler / build system. I am not sure where you guys get your inpression from: https://bun.sh/docs/bundler

**Comment by YummYume** [comment]
Bun is a JavaScript runtime designed to replace Node.js, but also a package manager, a test runner and a bundler. It aims at unifying everything a modern JavaScript developer needs in a project.

I agree with @stof that a Bun builder for Symfony should rather be a new package, though (and that it would need its own way of registering Stimulus controllers).

**Comment by phtmgt** [comment]
Yeah, I did not express my thoughts clearly enough. Should've said '... Bun is **also**  a [bun]dler / build...'.

**Comment by eokic** [comment]
So... why exactly is this "closed"?
